featuretools/primitives/standard/transform_primitive.py

```diff
@@ -1059,22 +1057,89 @@ class DateToHoliday(TransformPrimitive):
-            raise ValueError(error)
-        years_list = [1950 + x for x in range(150)]
-        self.federal_holidays = getattr(holidays, country)(years=years_list)
+        self.holidayUtil = HolidayUtil(country)
 
     def get_function(self):
         def date_to_holiday(x):
-            holidays_df = pd.DataFrame(sorted(self.federal_holidays.items()),
-                                       columns=['dates', 'names'])
-            holidays_df.dates = holidays_df.dates.astype('datetime64')
-            df = pd.DataFrame({'dates': x})
-            df.dates = df.dates.dt.normalize().astype('datetime64')
-            df = df.merge(holidays_df, how='left')
+            holiday_df = self.holidayUtil.to_df()
+            df = pd.DataFrame({'date': x})
+            df.date = df.date.dt.normalize().astype('datetime64')
+
+            df = df.merge(holiday_df, how='left', left_on='date', right_on='holiday_date')
             return df.names.values
         return date_to_holiday
+
+
+class DistanceToHoliday(TransformPrimitive):
+    """Computes the number of days before or after a given holiday.
+
+    Description:
+        For a list of dates, return the distance from the nearest
+        occurrence of a chosen holiday. The distance is returned in
+        days. If the closest occurrence is prior to the date given,
+        return a negative number.
+
+        If a date is missing, return `NaN`.
+
+        Currently only works with dates between 1950 and 2100.
+
+    Args:
+        holiday (str): Name of the holiday. Defaults to New Year's Day.
+
+        country (str): Specifies which country's calendar to use for the
+            given holiday. Default is `US`.
+
+    Examples:
+        >>> from datetime import datetime
+        >>> distance_to_holiday = DistanceToHoliday("New Year's Day")
+        >>> dates = [datetime(2010, 1, 1),
+        ...          datetime(2012, 5, 31),
+        ...          datetime(2017, 7, 31),
+        ...          datetime(2020, 12, 31)]
+        >>> distance_to_holiday(dates).tolist()
+        [0, -151, 154, 1]
+
+        We can also control the country in which we're searching for
+            a holiday.
+
+        >>> distance_to_holiday = DistanceToHoliday("Victoria Day", country='Canada')
+        >>> dates = [datetime(2010, 1, 1),
+        ...          datetime(2012, 5, 31),
+        ...          datetime(2017, 7, 31),
+        ...          datetime(2020, 12, 31)]
+        >>> distance_to_holiday(dates).tolist()
+        [143, -10, -70, 144]
+    """
+    name = "distance_to_holiday"
+    input_types = [ColumnSchema(logical_type=Datetime)]
+    return_type = ColumnSchema(semantic_tags={'numeric'})
+    default_value = 0
+
+    def __init__(self, holiday="New Year's Day", country="US"):
+        self.country = country
+        self.holiday = holiday
+        self.holidayUtil = HolidayUtil(country)
+
+        available_holidays = list(set(self.holidayUtil.federal_holidays.values()))
```

Comment:
Do we need to do `self` here?
```suggestion
        self.available_holidays = list(set(self.holidayUtil.federal_holidays.values()))
```

Reply:
its not used outside of __init__ so I don't think so